src/Rx/Weno/file_download.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/**
+ * Weno Zipped file download
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Kofi Appiah <kkappiah@medsov.com>
+ * @copyright Copyright (c) 2023 omega systems group international <info@omegasystemsgroup.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once("../../../interface/globals.php");
+
+use OpenEMR\Common\Crypto\CryptoGen;
+use OpenEMR\Rx\Weno\LogDataInsert;
+use OpenEMR\Common\Logging\EventAuditLogger;
+
+$cryptogen = new CryptoGen();
+$weno_username = $GLOBALS['weno_admin_username'];
+$weno_password = $GLOBALS['weno_encryption_key'];
+$encryption_key = $cryptoGen->decryptStandard($GLOBALS['weno_encryption_key']);
+$baseurl = "https://online.wenoexchange.com/en/EPCS/DownloadPharmacyDirectory";
+
+$data = array(
+    "UserEmail"             => $weno_username,
+    "MD5Password"           => md5($weno_password),
+    "ExcludeNonWenoTest"    => "N",
+    "Daily"                 => "Y"
+);
+
+if (date("l") == "Sunday") { //if today is Sunday download the weekly file
+    $data["Daily"] = "N";
+}
+
+$json_object = json_encode($data);
+$method = 'aes-256-cbc';
+
+$key = substr(hash('sha256', $encryption_key, true), 0, 32);
+
+$iv = chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0);
+
+$encrypted = base64_encode(openssl_encrypt($json_object, $method, $key, OPENSSL_RAW_DATA, $iv));
+
+$fileUrl = $baseurl . "?useremail=" . $weno_username . "&data=" . urlencode($encrypted);
+$storelocation = $GLOBALS['fileroot'] . "/contrib/weno/weno_pharmacy.zip";
+$path_to_extract = $GLOBALS['fileroot'] . "/contrib/weno/";
```

Comment:
Note users will not have write privileges to this directory. Recommend using php temp directory to do all this that then remove when the script is completed.